returnn/tf/util/data.py

```diff
@@ -2337,6 +2337,32 @@ def copy_extend_batch(self, batch):
         data.placeholder = x
     return data
 
+  def copy_unpack_batch(self):
+    """
+    If the batch dim contains a :class:`BatchInfo.PackedDim`, unpack it and restore the data from before the packing.
+
+    :rtype: Data
+    """
+    assert self.have_batch_axis()
+    assert self.batch, "%s: batch unset" % self
+    data = self.copy()
+    kwargs = self.get_kwargs(include_special_axes=False)
+
+    dim_tags = []
+    for dim_tag in data.dim_tags:
+      if dim_tag.is_batch_dim() and dim_tag.batch and len(dim_tag.batch.virtual_dims) > 0:
```

Comment:
See my previous comment.
```suggestion
      if dim_tag.is_batch_dim() and len(dim_tag.batch.virtual_dims) > 0:
```
Ok, you can leave the `len(dim_tag.batch.virtual_dims) > 0` though. In that case, it makes sense to just leave it as-is.